### PR TITLE
teleport.cluster.local cleanup

### PR DESCRIPTION
**Pull request by Joerger**
From my understanding, the value `teleport.cluster.local` is used to create valid credentials that pass the TLS SNI check regardless of the actual DNS that the client is connects to. This logic can be confusing for new Teleporters (myself included) and is lacking comments in some spots. It was also being used in some places where it wasn't actually necessary.

This PR adds some comments and cleans up some of the `constants.APIDomain`. It also removes a couple sections of deprecated logic from before `v4.4`.

### CODE REVIEW

File: api/client/client.go [hunk 1]
```diff
@@ -325,7 +324,7 @@ func dialerConnect(ctx context.Context, params connectParams) (*Client, error) {
 		params.dialer = params.cfg.Dialer
 	}
 	clt := newClient(params.cfg, params.dialer, params.tlsConfig)
-	if err := clt.dialGRPC(ctx, constants.APIDomain); err != nil {
+	if err := clt.dialGRPC(ctx, ""); err != nil {
```
Comment (zmb3) [inline api/client/client.go:327]: Overall I find this PR very helpful in understanding what's going on.

This line, however, might still leave me wondering what's going on - any way we can clarify what it means to dial an empty addr?

Comment (Joerger) [reply api/client/client.go:333]: Added coments

File: api/client/client.go [hunk 2]
```diff
@@ -283,6 +283,7 @@ type connectParams struct {
 	sshConfig *ssh.ClientConfig
 }
 
+// proxyConnect connects to the Teleport Auth Server directly.
```
Comment (zmb3) [inline api/client/client.go:286]:
```suggestion
// authConnect connects to the Teleport Auth Server directly.
```

File: api/client/client.go [hunk 3]
```diff
@@ -292,6 +293,7 @@ func authConnect(ctx context.Context, params connectParams) (*Client, error) {
 	return clt, nil
 }
 
+// proxyConnect connects to the Teleport Auth Server through the reverse tunnel proxy.
```
Comment (zmb3) [inline api/client/client.go:296]:
```suggestion
// tunnelConnect connects to the Teleport Auth Server through the proxy's reverse tunnel.
```

**Review by andrejtokarcik — APPROVED** [review]

**Review by tcsc — APPROVED** [review]

**Comment by Joerger** [comment]
@zmb3 can you re-review?

**Review by zmb3 — APPROVED** [review]

**Review by r0mant — COMMENTED** [review]
@Joerger Left a couple of comments here. They don't show up on the bottom for some reason, probably cause I commented on already existing threads.

**Pull request merged**